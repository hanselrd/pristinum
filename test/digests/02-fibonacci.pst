func fib(n @u64) @u64:
    if n < 2:
        return n;
    end
    return fib(n - 1) + fib(n - 2);
end

func main() @void:
    n @u64 := 1;
    while n < 15:
        print(fib(n));
        ++n;
    end
end