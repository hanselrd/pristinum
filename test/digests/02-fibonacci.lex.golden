[ PT
    ( Pn 0 1 1 )
    ( TK
        ( TokSymbol
            { tsText = "func"
            , tsID = 44
            }
        )
    )
, PT
    ( Pn 5 1 6 )
    ( T_IDENT "fib" )
, PT
    ( Pn 8 1 9 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 9 1 10 )
    ( T_IDENT "n" )
, PT
    ( Pn 11 1 12 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 12 1 13 )
    ( TK
        ( TokSymbol
            { tsText = "u64"
            , tsID = 57
            }
        )
    )
, PT
    ( Pn 15 1 16 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 17 1 18 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 18 1 19 )
    ( TK
        ( TokSymbol
            { tsText = "u64"
            , tsID = 57
            }
        )
    )
, PT
    ( Pn 21 1 22 )
    ( TK
        ( TokSymbol
            { tsText = ":"
            , tsID = 18
            }
        )
    )
, PT
    ( Pn 27 2 5 )
    ( TK
        ( TokSymbol
            { tsText = "if"
            , tsID = 49
            }
        )
    )
, PT
    ( Pn 30 2 8 )
    ( T_IDENT "n" )
, PT
    ( Pn 32 2 10 )
    ( TI "2" )
, PT
    ( Pn 34 2 12 )
    ( TK
        ( TokSymbol
            { tsText = "<"
            , tsID = 21
            }
        )
    )
, PT
    ( Pn 35 2 13 )
    ( TK
        ( TokSymbol
            { tsText = "``"
            , tsID = 34
            }
        )
    )
, PT
    ( Pn 37 2 15 )
    ( TK
        ( TokSymbol
            { tsText = ":"
            , tsID = 18
            }
        )
    )
, PT
    ( Pn 47 3 9 )
    ( TK
        ( TokSymbol
            { tsText = "return"
            , tsID = 51
            }
        )
    )
, PT
    ( Pn 54 3 16 )
    ( T_IDENT "n" )
, PT
    ( Pn 55 3 17 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 61 4 5 )
    ( TK
        ( TokSymbol
            { tsText = "end"
            , tsID = 40
            }
        )
    )
, PT
    ( Pn 69 5 5 )
    ( TK
        ( TokSymbol
            { tsText = "return"
            , tsID = 51
            }
        )
    )
, PT
    ( Pn 76 5 12 )
    ( T_IDENT "fib" )
, PT
    ( Pn 79 5 15 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 80 5 16 )
    ( T_IDENT "n" )
, PT
    ( Pn 82 5 18 )
    ( TI "1" )
, PT
    ( Pn 84 5 20 )
    ( TK
        ( TokSymbol
            { tsText = "-"
            , tsID = 13
            }
        )
    )
, PT
    ( Pn 85 5 21 )
    ( TK
        ( TokSymbol
            { tsText = "``"
            , tsID = 34
            }
        )
    )
, PT
    ( Pn 87 5 23 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 89 5 25 )
    ( T_IDENT "fib" )
, PT
    ( Pn 92 5 28 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 93 5 29 )
    ( T_IDENT "n" )
, PT
    ( Pn 95 5 31 )
    ( TI "2" )
, PT
    ( Pn 97 5 33 )
    ( TK
        ( TokSymbol
            { tsText = "-"
            , tsID = 13
            }
        )
    )
, PT
    ( Pn 98 5 34 )
    ( TK
        ( TokSymbol
            { tsText = "``"
            , tsID = 34
            }
        )
    )
, PT
    ( Pn 100 5 36 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 102 5 38 )
    ( TK
        ( TokSymbol
            { tsText = "+"
            , tsID = 10
            }
        )
    )
, PT
    ( Pn 103 5 39 )
    ( TK
        ( TokSymbol
            { tsText = "``"
            , tsID = 34
            }
        )
    )
, PT
    ( Pn 105 5 41 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 107 6 1 )
    ( TK
        ( TokSymbol
            { tsText = "end"
            , tsID = 40
            }
        )
    )
, PT
    ( Pn 112 8 1 )
    ( TK
        ( TokSymbol
            { tsText = "func"
            , tsID = 44
            }
        )
    )
, PT
    ( Pn 117 8 6 )
    ( T_IDENT "main" )
, PT
    ( Pn 121 8 10 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 122 8 11 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 124 8 13 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 125 8 14 )
    ( TK
        ( TokSymbol
            { tsText = "void"
            , tsID = 60
            }
        )
    )
, PT
    ( Pn 129 8 18 )
    ( TK
        ( TokSymbol
            { tsText = ":"
            , tsID = 18
            }
        )
    )
, PT
    ( Pn 135 9 5 )
    ( T_IDENT "n" )
, PT
    ( Pn 137 9 7 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 138 9 8 )
    ( TK
        ( TokSymbol
            { tsText = "u64"
            , tsID = 57
            }
        )
    )
, PT
    ( Pn 142 9 12 )
    ( TK
        ( TokSymbol
            { tsText = ":="
            , tsID = 19
            }
        )
    )
, PT
    ( Pn 145 9 15 )
    ( TI "1" )
, PT
    ( Pn 146 9 16 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 152 10 5 )
    ( TK
        ( TokSymbol
            { tsText = "while"
            , tsID = 61
            }
        )
    )
, PT
    ( Pn 158 10 11 )
    ( T_IDENT "n" )
, PT
    ( Pn 160 10 13 )
    ( TI "15" )
, PT
    ( Pn 163 10 16 )
    ( TK
        ( TokSymbol
            { tsText = "<"
            , tsID = 21
            }
        )
    )
, PT
    ( Pn 164 10 17 )
    ( TK
        ( TokSymbol
            { tsText = "``"
            , tsID = 34
            }
        )
    )
, PT
    ( Pn 166 10 19 )
    ( TK
        ( TokSymbol
            { tsText = ":"
            , tsID = 18
            }
        )
    )
, PT
    ( Pn 176 11 9 )
    ( T_IDENT "print" )
, PT
    ( Pn 181 11 14 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 182 11 15 )
    ( T_IDENT "fib" )
, PT
    ( Pn 185 11 18 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 186 11 19 )
    ( T_IDENT "n" )
, PT
    ( Pn 187 11 20 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 188 11 21 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 189 11 22 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 199 12 9 )
    ( T_IDENT "n" )
, PT
    ( Pn 201 12 11 )
    ( TK
        ( TokSymbol
            { tsText = "="
            , tsID = 24
            }
        )
    )
, PT
    ( Pn 203 12 13 )
    ( T_IDENT "n" )
, PT
    ( Pn 205 12 15 )
    ( TK
        ( TokSymbol
            { tsText = "++"
            , tsID = 11
            }
        )
    )
, PT
    ( Pn 207 12 17 )
    ( TK
        ( TokSymbol
            { tsText = "`"
            , tsID = 33
            }
        )
    )
, PT
    ( Pn 208 12 18 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 214 13 5 )
    ( TK
        ( TokSymbol
            { tsText = "end"
            , tsID = 40
            }
        )
    )
, PT
    ( Pn 218 14 1 )
    ( TK
        ( TokSymbol
            { tsText = "end"
            , tsID = 40
            }
        )
    )
]